AX: Move accessibilityThreadTextApisEnabled to DeprecatedGlobalSettings

hoffmanjoshua opened the pull request:
#### 18aeb0e661445201b72923ce20216bc52033b2a3
<pre>
AX: Move accessibilityThreadTextApisEnabled to DeprecatedGlobalSettings
<a href="https://bugs.webkit.org/show_bug.cgi?id=281031">https://bugs.webkit.org/show_bug.cgi?id=281031</a>
<a href="https://rdar.apple.com/137469268">rdar://137469268</a>

Reviewed by Tyler Wilcock.

`accessibilityThreadTextApis` tracks global state. Currently, we store this in the Document&apos;s
settings, but we can&apos;t guarantee the value is the same across different documents. Instead,
let&apos;s move it to `DeprecatedGlobalSettings`, and store/access it on the secondary thread using
the existing `gAccessibiltiyThreadTextApisEnabled` atomic bool.

This patch also includes a drive-by build fix for the new `nextUnignoredSiblingOrParent` when
`ENABLE_AX_THREAD_TEXT_APIS` is enabled. These calls were missing the `shouldUpdateChildren`
parameter, which should be true to match the previous behavior.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::nextUnignoredSiblingOrParent const):
* Source/WebCore/accessibility/AXCoreObject.h:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::AXObjectCache):
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setAccessibilityThreadTextApisEnabled):
(WebCore::DeprecatedGlobalSettings::accessibilityThreadTextApisEnabled):

Canonical link: <a href="https://commits.webkit.org/284827@main">https://commits.webkit.org/284827@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/62a2c8ef12c9f4ed402ce59e2e7dca447d0c52ff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/70611 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50017 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23376 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/74700 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/21789 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/72728 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57817 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/21629 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55923 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14397 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/73677 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45478 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60869 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36385 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42135 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/18305 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/20150 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/63732 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64067 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18660 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76420 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/69859 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/14837 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17869 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63656 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/14880 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60937 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63597 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15640 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11638 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5285 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/91641 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/45819 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/587 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/19978 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/46893 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48170 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/46635 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on Source/WebCore/accessibility/AXTextMarker.cpp:
```diff
@@ -536,7 +536,7 @@ CharacterRange AXTextMarker::characterRangeForLine(unsigned lineIndex) const
     RELEASE_ASSERT(!offset());
 
-    auto* stopObject = object->nextUnignoredSiblingOrParent();
+    auto* stopObject = object->nextUnignoredSiblingOrParent(true);
```
Thank you for fixing my mistake! I wonder if we should just remove the parameter outright, since no existing caller needs to pass a `false` 🤔

hoffmanjoshua committed:
AX: Move accessibilityThreadTextApisEnabled to DeprecatedGlobalSettings
https://bugs.webkit.org/show_bug.cgi?id=281031
rdar://137469268

Reviewed by Tyler Wilcock.

`accessibilityThreadTextApis` tracks global state. Currently, we store this in the Document's
settings, but we can't guarantee the value is the same across different documents. Instead,
let's move it to `DeprecatedGlobalSettings`, and store/access it on the secondary thread using
the existing `gAccessibiltiyThreadTextApisEnabled` atomic bool.

This patch also includes a drive-by build fix for the new `nextUnignoredSiblingOrParent` when
`ENABLE_AX_THREAD_TEXT_APIS` is enabled. These calls were missing the `shouldUpdateChildren`
parameter, which should be true to match the previous behavior.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::nextUnignoredSiblingOrParent const):
* Source/WebCore/accessibility/AXCoreObject.h:
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::AXObjectCache):
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setAccessibilityThreadTextApisEnabled):
(WebCore::DeprecatedGlobalSettings::accessibilityThreadTextApisEnabled):

Canonical link: https://commits.webkit.org/284827@main
